app/web/src/helpers/cadPackages/openScadController.js

```diff
@@ -41,51 +46,61 @@ export const render = async ({ code, settings }) => {
     })
     if (response.status === 400) {
       const { error } = await response.json()
-      const cleanedErrorMessage = error.replace(
-        /["|']\/tmp\/.+\/main.scad["|']/g,
-        "'main.scad'"
-      )
-      return {
-        status: 'error',
-        message: {
-          type: 'error',
-          message: cleanedErrorMessage,
-          time: new Date(),
-        },
-      }
+      const cleanedErrorMessage = cleanError(error)
+      return createUnhealthyResponse(new Date(), cleanedErrorMessage)
     }
     const data = await response.json()
-    return {
-      status: 'healthy',
-      objectData: {
-        type: 'png',
-        data: data.url,
-      },
-      message: {
-        type: 'message',
-        message: data.consoleMessage,
-        time: new Date(),
-      },
-    }
+    const type = data.type !== 'stl' ? 'png' : 'geometry'
+    const newData = data.type !== 'stl' ? data.url : stlToGeometry(data.url)
```

Comment:
Previously we were storing the aws url here and than <Asset /> was downloading and managing the geometry, but instead doing that here, it's much easier to grab that geometry again to download the stl